### PR TITLE
Add 0.204 release notes

### PR DESCRIPTION
**Pull request by martint**
Initial set of notes from https://github.com/prestodb/presto/issues/10742. Some items need rephrasing.

### CODE REVIEW

**Comment by electrum** [comment]
Additions:
```
Hive Connector Changes
----------------------

* Enable optimized Parquet reader and predicate pushdown by default.
* Disallow creating tables with unsupported partition types.
* Fix memory accounting for sort buffer.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Web UI Changes
+--------------
+* Display resource estimates in Web UI query details page.
+
+Resource Group Changes
+----------------------
+* Fix a bug that caused unnecessary queuing in deployments where no resource group configuration was specified.
+
+Hive Changes
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: We started using "XXX Connector Changes" for all of these. It eliminates the choice of which connectors to say "Connector" for, avoids ambiguity for things like "JMX Changes", and matches the language we use elsewhere in the documentation.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+------------
+* Disallow creating tables with unsupported partition types.
+* Support overwriting partitions for insert queries. This behavior is controlled by session property ``insert_existing_partitions_behavior``.
+* Fix an issue that causes optimized ORC writer to write stripes with excessive size for highly dictionary encoded columns.
+
+JMX Changes
+-----------
+* Track execution statistics of `AddExchanges` and `PredicatePushdown` optimizer rules.
+
+Cassandra Changes
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Cassandra Connector Changes

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
+=============
+Release 0.203
+=============
+
+General
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: General Changes

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+-------
+* Fix a memory tracking bug that causes the memory to be overestimated for scan operators when reading ORC files.
+* Use distributed join if one side is naturally partitioned on join keys.
+* Improve performance of correlated subqueries when filters from outer query can be propagated to the subquery.
+* Improve performance for correlated subqueries that contain inequalities.
+* Add support for all geometry types in :func:`ST_Area`.
+* Add :func:`ST_EnvelopeAsPts` function.
+* Add :func:`to_big_endian_32` and :func:`from_big_endian_32` functions.
+* Add cast between ``VARBINARY`` type and ``IPADDRESS`` type.
+* Make :func:`lpad` and :func:`rpad` functions support varbinary in addition to varchar.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Code formatting and uppercase for type names.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+* Fill in with ``NULL`` when processing arrays of different length in :func:`zip_with`.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: How about
```
* Allow using arrays of mismatched lengths with :func:`zip_with`.
  The missing positions are filled with ``NULL``.
```
Also, fix the documentation for `zip_with` in `array.rst`. The "NULL" part only has single backticks but needs to have double.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
+=============
+Release 0.203
+=============
+
+General
+-------
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Newline after section header

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+* Fix a memory tracking bug that causes the memory to be overestimated for scan operators when reading ORC files.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: The engine doesn't know anything about ORC files. Phrase it like this:
```
* Fix a memory tracking bug that causes the memory to be overestimated for scan operators.
  In particular, this affects reading ORC files in the Hive connector.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+* Make :func:`lpad` and :func:`rpad` functions support varbinary in addition to varchar.
+* Fill in with ``NULL`` when processing arrays of different length in :func:`zip_with`.
+
+Event Listener Changes
+----------------------
+* Add resource estimates to query events.
+
+Web UI Changes
+--------------
+* Display resource estimates in Web UI query details page.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: "Web UI" is redundant here

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+Resource Group Changes
+----------------------
+* Fix a bug that caused unnecessary queuing in deployments where no resource group configuration was specified.
+
+Hive Changes
+------------
+* Disallow creating tables with unsupported partition types.
+* Support overwriting partitions for insert queries. This behavior is controlled by session property ``insert_existing_partitions_behavior``.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Let's document the supported values for the session property.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+* Fix an issue that causes optimized ORC writer to write stripes with excessive size for highly dictionary encoded columns.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]:
```
* Prevent the optimized ORC writer from writing excessively large stripes for
  highly compressed, dictionary encoded columns.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+JMX Changes
+-----------
+* Track execution statistics of `AddExchanges` and `PredicatePushdown` optimizer rules.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Double backticks

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.204.rst]: This change seems to be an engine JMX stats change, not a JMX Connector change, so move this to General.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+Cassandra Changes
+-----------------
+* Add support for read from materialized views.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]:
```
* Add support for reading from materialized views.

File: presto-docs/src/main/sphinx/release/release-0.204.rst (hunk 7 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-0.204.rst]: I think this is an issue specifically for ORC, not for scan operators. So, if anything, it's the opposite -- the bug in ORC affects the memory estimation for the scan operator.

Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.204.rst]: Yes, this is specifically for ORC files, not scans in general.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+Event Listener Changes
+----------------------
+
+* Add resource estimates to query events.
+
+UI Changes
+----------
+
+* Display resource estimates in Web UI query details page.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Addition:
```
* Fix kill query button.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+* Allow using arrays of mismatched lengths with :func:`zip_with`.
+  The missing positions are filled with ``NULL``.
+* Track execution statistics of ``AddExchanges`` and ``PredicatePushdown`` optimizer rules.
+
+Event Listener Changes
+----------------------
+
+* Add resource estimates to query events.
+
+UI Changes
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]:
```
Web UI Changes
--------------

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+Resource Group Changes
+----------------------
+
+* Fix a bug that caused unnecessary queuing in deployments where no resource
+  group configuration was specified.
+
+Hive Connector Changes
+----------------------
+
+* Fix a memory tracking bug that causes the memory to be overestimated for scan
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: "that caused"

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: How about
```
* Fix over-estimation of memory usage for scan operators when reading ORC files.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+UI Changes
+----------
+
+* Display resource estimates in Web UI query details page.
+
+Resource Group Changes
+----------------------
+
+* Fix a bug that caused unnecessary queuing in deployments where no resource
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: We can remove the "a bug" part:
```
* Fix unnecessary queuing in deployments where no resource group configuration was specified.

File: presto-docs/src/main/sphinx/release/release-0.204.rst [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+* Fix a memory tracking bug that causes the memory to be overestimated for scan
+  operators when reading ORC files.
+* Disallow creating tables with unsupported partition types.
+* Support overwriting partitions for insert queries. This behavior is controlled
+  by session property ``insert_existing_partitions_behavior``.
+* Prevent the optimized ORC writer from writing excessively large stripes for
+  highly compressed, dictionary encoded columns.
+* Enable optimized Parquet reader and predicate pushdown by default.
+* Fix memory accounting for sort buffer used for writing sorted bucketed tables.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.204.rst]: Move to top with other fix

**Pull request closed without merging**